spec/controllers/catalog_controller_spec.rb

```diff
@@ -32,5 +32,47 @@
         expect(controller).to have_search_parameters
       end
     end
+
+    context "when not logged in" do
```

Comment:
Just thinking big picture here... Testing this stuff through controllers is leaning on the context of "current user" and "logged in". Along the lines I outlined above, I wonder if breaking the binding to user session clarifies the architecture and business rules. The convenience pieces of passing along the current user can be applied easily.

That is... Are these controller specs really absorbing a deficiency in testability by adding coupling? Could the persister/index rules be tested statelessly (by passing a user)? Could the controller specs be checked against a mock persister/index just to verify that they are calling the right stuff (doing the HTTP param bits as expected and asking for items as the expected user)?